pkg/network/nr/container.go

```diff
@@ -177,17 +177,17 @@ func (nr *NetResource) Leave(containerID string) error {
 		Str("container", containerID).
 		Msg("delete container network namespace")
 
-	ns, err := namespace.GetByName(containerID)
-	if err != nil && !os.IsNotExist(err) {
-		return err
-	}
-	if os.IsNotExist(err) {
-		// nothing to do, early return
+	namespc, err := namespace.GetByName(containerID)
+	if _, ok := err.(ns.NSPathNotExistErr); ok {
```

Comment:
The proper way to do this these days is with `errors.As(...)`, since the error itself does not seem to be used anyhow you might even be able to get away with `errors.Is`, though I'm not sure about that

Reply:
Tried both those options but I did not get that to work..

Reply:
That's rather strange since the documentations of errors.As here (https://golang.org/pkg/errors/) literally explains that `As` should be preferred over this construct

Reply:
Ah but wait, I used a pointer while it should have been a concrete type.